File: ATLASDEV/AtlasModel/Enterprise/TTQueryDef.cs
using AtlasModel.Enterprise.Enums;
using System;

namespace AtlasModel.Enterprise
{
    public partial class TTQueryDef
    {
        public Guid QueryDefId { get; set; }
        public Guid ObjectDefId { get; set; }
        public string Name { get; set; }
        public string Nql { get; set; }
        public QueryDefSubTypeEnum SubType { get; set; }
        public Guid ModuleDefId { get; set; }
        public string CheckOutId { get; set; }
        public short CheckOutStatus { get; set; }
        public string Methods { get; set; }
        public string Description { get; set; }
    }
}

File: AtlasWeb/Core/wwwroot/app/NebulaClient/StorageManager/DefinitionManagement/TTQueryDef.ts
import { Guid } from "../../Mscorlib/Guid";
import { TTModuleDef } from "./TTModuleDef";
import { TTObjectDef } from "./TTObjectDef";
import { QueryDefSubTypeEnum } from "../../Utils/Enums/QueryDefSubTypeEnum";
import { TTDefCollection } from "./TTDefCollection";
import { TTQueryParameterDef } from "./TTQueryParameterDef";

export class TTQueryDef {
    public ModuleDefID: Guid;
    public ModuleDef: TTModuleDef;
    public ID: Guid;
    public static INJECTION_NQL_PLACE_HOLDER: string = "$B$";
    public ObjectDef: TTObjectDef;
    public ObjectDefID: Guid;
    public QueryDefID: Guid;
    public Name: string;
    public StaticFunctionName: string;
    public ClassName: string;
    public FullClassName: string;
    public Nql: string;
    public Description: string;
    public Methods: string;
    public Subtype: QueryDefSubTypeEnum;
    public ParameterDefs: TTDefCollection<TTQueryParameterDef>;
    public ToString(): string {
        return null;
    }
}
